src/utils/text/index.js

```diff
@@ -76,18 +76,27 @@ export const getTextForLabel = (
     accession: string;
     name: string;
     short_name: string;
+    source_database: string;
   } */,
   label /*: {
     accession: boolean;
     name: boolean;
     short: boolean;
   } */,
 ) /*: string */ => {
-  let text = label.short ? entry.short_name || '' : '';
-  if (text.length > 0 && label.accession) text += ' - ';
-  if (label.accession) text += entry.accession;
-  if (text.length > 0 && label.name) text += ': ';
-  if (label.name) text += entry.name;
-  if (text.length === 0) text += entry.accession;
+  let text = '';
+
+  if (!entry.accession.startsWith('IPR')) {
+    text = entry.source_database.toUpperCase() + ': ';
+  }
+
+  if (entry.short_name) text += entry.short_name;
+  else if (entry.name) text += entry.name;
+  else if (entry.accession) text += entry.accession;
+
+  if (text.length > 20) {
+    text = text.slice(0, 20) + '..';
+  }
```

Comment:
Instead of truncating the text, add the accession if both the short name and long name aren't available.